files/scripts/arp_update

```diff
@@ -89,32 +89,57 @@ while /bin/true; do
       eval `eval $ip6cmd`
  
       if [[ $SUBTYPE == "dualtor" ]]; then
-        # manually set any remaining FAILED/INCOMPLETE entries to permanently INCOMPLETE
-        # this prevents any remaining INCOMPLETE entries from automatically transitioning to FAILED
-        # once these entries are incomplete, any subsequent neighbor advertisement messages
-        # are able to resolve the entry
-
-        # generates the following command for each failed or incomplete IPv6 neighbor
-        # ip neigh replace <neighbor IPv6> dev <VLAN name> nud incomplete
-        neigh_replace_template="sed -e 's/^/ip neigh replace /' -e 's/,/ dev /' -e 's/$/ nud incomplete;/'" 
-        ip_neigh_replace_cmd="ip -6 neigh show | grep -v fe80 | grep $vlan | grep -E 'FAILED|INCOMPLETE' | cut -d ' ' -f 1,3 --output-delimiter=',' | $neigh_replace_template"
-        eval `eval $ip_neigh_replace_cmd`
-
-        # on dual ToR devices, try to resolve failed neighbor entries since
-        # these entries will have tunnel routes installed, preventing normal 
-        # neighbor resolution (SWSS PR #2137)
-
-        # since ndisc6 is a userland process, the above ndisc6 commands are 
-        # insufficient to update the kernel neighbor table for failed entries
-
-        # we don't need to do this for ipv4 neighbors since arping is able to
-        # update the kernel neighbor table
-
-        # generates the following command for each failed or incomplete IPv6 neighbor
+        # capture all current failed/incomplete IPv6 neighbors in the kernel to avoid situations where new neighbors are learned
+        # in the middle of the below sequence of commands
+        unresolved_kernel_neighbors=$(ip -6 neigh show | grep -v fe80 | grep $vlan | grep -E 'FAILED|INCOMPLETE')
+        failed_kernel_neighbors=$(echo "$unresolved_kernel_neighbors" | grep FAILED | cut -d ' ' -f 1)
+
+        # it's possible for kernel neighbors to fall out of sync with the hardware
+        # this can result in failed neighbors entries that don't have corresponding zero MAC neighbor entries
+        # and therefore don't have tunnel routes installed in the hardware
+        # flush these neighbors from the kernel to force relearning and resync them to the hardware:
+        # 1. for every FAILED or INCOMPLETE neighbor in the kernel, check if there is a corresponding zero MAC neighbor in APPL_DB
+        # 2. if no zero MAC neighbor entry exists, flush the kernel neighbor entry
+        #     - generates the command 'ip neigh flush <neighbor IPv6>' for all such neighbors
```

Comment:
isn't `dev VlanX` required for flush?

Reply:
no, the command still runs without specifying the vlan